### DISCUSSION TITLE
Is it possible to disable logging?

### DISCUSSION THREAD

**Discussion by crazy25000** [nautechsystems/nautilus_trader#344] [Q&A] [answered]
Hello, 

Is it possible to disable logging? Useful to disable when running parameterized backtests and only needing the strategy metrics. I've tried:
- setting `BacktestEngine(bypass_logging=True, level_stdout=LogLevel.ERRORS)`
- passing config data with same settings as above, but it doesn't disable it

Related doc: https://docs.nautilustrader.io/api_reference/backtest.html#nautilus_trader.backtest.engine.BacktestEngine 

Thanks!

**Comment by cjdsellers** [comment, marked as answer]
I just tested the bypass on latest `master` and `develop`, it is indeed bypassing.

Note that it'll still log the system and backtest specs, diagnostics and reports - however for performance reasons it won't log anything in the backtest iterations where the biggest performance hits would be taken.

**Reply by crazy25000** [reply]
I see, it's the specs I was still getting. Thanks.

**Reply by cjdsellers** [reply]
Ah yes, you do make a good point actually. With logging bypassed it shouldn't run that entire backtest footer sequence, but instead leave it to the user to pull out any metrics (or not), before running again. 

I'll make that change soon.

**Reply by crazy25000** [reply]
Awesome, thank you!

**Reply by cjdsellers** [reply]
The next push to develop adds a `run_analysis` option flag to the `BacktestEngine` config. This controls whether post backtest performance analysis is run - some users may want to skip this step for performance reasons, or conduct their own custom analysis.

The `bypass_logging`  still only controls whether logging is bypassed during the backtest run iterations, as the small amount of logging which occurs in the pre and post back test run sequences would have nominal performance overhead.

**Reply by crazy25000** [reply]
Can't wait to test it out! :D
